actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -376,13 +376,11 @@ def multiple_sources_tag_builder(type, sources)
         end
 
         def resolve_image_source(source, skip_pipeline)
-          if source.is_a?(Symbol) || source.is_a?(String)
-            path_to_image(source, skip_pipeline: skip_pipeline)
-          else
+          if source.respond_to? :to_model
             polymorphic_url(source)
+          else
+            path_to_image(source, skip_pipeline: skip_pipeline)
```

Comment:
There are more callers of `path_to_image` (& friends) nearby... should those also be passing through this method?

(srcset handling inside image_tag; poster + actual sources inside video_tag, ...)